cmd/options.go

```diff
@@ -235,6 +236,18 @@ func getOptions(flags *pflag.FlagSet) (lib.Options, error) {
 		opts.ConsoleOutput = null.StringFrom(redirectConFile)
 	}
 
+	opts.DNS = &lib.DNSConfig{}
+	if dns, err := flags.GetString("dns"); err != nil {
+		return opts, err
+	} else if dns != "" {
+		if err := opts.DNS.UnmarshalText([]byte(dns)); err != nil {
+			return opts, err
+		}
+		// TODO: This obviously can't work for neither option...
+		opts.DNS.TTL.Valid = flags.Changed("dns")
+		opts.DNS.Strategy.Valid = flags.Changed("dns")
+	}
```

Comment:
I think you could try to implement [pflag.Value](https://godoc.org/github.com/spf13/pflag#Value) for example see [here](https://github.com/spf13/pflag/blob/81378bbcd8a1005f72b1e8d7579e5dd7b2d612ab/bytes.go#L11-L34)

Reply:
Don't think this is needed anymore, since it seems to work now after switching to a struct value instead of a pointer for `DNSConfig`.